puppet-enterprise-cluster/azuredeploy.json

```diff
@@ -0,0 +1,520 @@
+      "type": "Microsoft.Compute/virtualMachines",
+      "name": "[concat('UbuntuAgentVM',copyIndex())]",
+      "location": "[resourceGroup().location]",
+      "copy": {
+        "name": "ubuntuAgentVmLoop",
+        "count": "[parameters('linuxAgentCount')]"
+      },
+      "dependsOn": [
+        "[concat('Microsoft.Storage/storageAccounts/', variables('newStorageAccountName'))]",
+        "[concat('Microsoft.Network/networkInterfaces/', 'peAgentNic', add(copyIndex(),parameters('windowsAgentCount')))]"
+      ],
+      "properties": {
+        "hardwareProfile": {
+          "vmSize": "[parameters('vmSize')]"
+        },
+        "osProfile": {
+          "computername": "[concat('UbuntuAgentVM',copyIndex())]",
+          "adminUsername": "[parameters('adminUsername')]",
+          "adminPassword": "[parameters('adminPassword')]"
+        },
+        "storageProfile": {
+          "imageReference": {
+            "publisher": "canonical",
+            "offer": "ubuntuserver",
+            "sku": "14.04.2-LTS",
+            "version": "latest"
+          },
+          "osDisk": {
+            "name": "osdisk",
+            "vhd": {
+              "uri": "[concat('http://',variables('newStorageAccountName'),'.blob.core.windows.net/vhds/','osdisk',add(copyIndex(),parameters('windowsAgentCount')),'.vhd')]"
+            },
+            "caching": "ReadWrite",
+            "createOption": "FromImage"
+          }
+        },
+        "networkProfile": {
+          "networkInterfaces": [
+            {
+              "id": "[resourceId('Microsoft.Network/networkInterfaces',concat('peAgentNic',add(copyIndex(),parameters('windowsAgentCount'))))]"
+            }
+          ]
+        },
+        "diagnosticsProfile": {
+          "bootDiagnostics": {
+            "enabled": "true",
+            "storageUri": "[concat('http://',variables('newStorageAccountName'),'.blob.core.windows.net')]"
+          }
+        }
+      }
+    },
+    {
+        "type": "Microsoft.Compute/virtualMachines/extensions",
+        "name": "[concat('UbuntuAgentVM',copyIndex(),'/CustomScriptExtension')]",
+        "apiVersion": "2016-03-30",
+        "location": "[resourceGroup().location]",
+        "copy": {
+          "name": "linuxAgentExtLoop",
+          "count": "[parameters('linuxAgentCount')]"
+        },
+        "dependsOn": [
+            "[concat('Microsoft.Compute/virtualMachines/', concat('UbuntuAgentVm',copyIndex()))]",
+            "[concat('Microsoft.Compute/virtualMachines/', variables('puppetMasterVmName'),'/extensions/CustomScriptExtension')]"
+        ],
+        "properties": {
+           "publisher": "Microsoft.OSTCExtensions",
+           "type": "CustomScriptForLinux",
+           "typeHandlerVersion": "1.3",
+           "autoUpgradeMinorVersion" : "true",
+           "settings": {
+              "fileUris": ["https://raw.githubusercontent.com/azure/azure-quickstart-templates/master/puppet-enterprise-cluster/install_puppet_agent.sh"],
```

Comment:
repo name should be Azure. The CI can replace URLs to your repo automatically https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#travis-ci